src/main/scala/org/antipathy/scoozie/action/FsAction.scala

```diff
@@ -92,30 +93,34 @@ object FsAction {
     * Create a new instance of this action from a configuration
     */
   def apply(config: Config): Node =
-    try {
-      FsAction(name = config.getString("name"),
-               actions = buildFSSteps(config.getConfigList("steps")),
-               jobXmlOption = if (config.hasPath("job-xml")) {
-                 Some(config.getString("job-xml"))
+    Try {
+      FsAction(name = config.getString(HoconConstants.name),
+               actions = buildFSSteps(config.getConfigList(HoconConstants.steps)),
+               jobXmlOption = if (config.hasPath(HoconConstants.jobXml)) {
+                 Some(config.getString(HoconConstants.jobXml))
                } else None,
                configuration = ConfigurationBuilder.buildConfiguration(config))
-    } catch {
-      case c: ConfigException =>
-        throw new ConfigurationMissingException(s"${c.getMessage} in ${config.getString("name")}")
+    } match {
+      case Success(node) => node
+      case Failure(exception) =>
+        throw new ConfigurationMissingException(s"${exception.getMessage} in ${config.getString(HoconConstants.name)}")
     }
 
   /**
     * Build the steps in this action from a collection of configuration objects
     */
   private def buildFSSteps(configList: util.List[_ <: Config]): Seq[FileSystemAction] =
     Seq(configList.asScala.map {
-      case delete if delete.hasPath("delete") => Delete(delete.getString("delete"))
-      case mkdir if mkdir.hasPath("mkdir")    => MakeDir(mkdir.getString("mkdir"))
-      case touchz if touchz.hasPath("touchz") => Touchz(touchz.getString("touchz"))
-      case chmod if chmod.hasPath("chmod") =>
-        Chmod(chmod.getString("chmod.path"), chmod.getString("chmod.permissions"), chmod.getString("chmod.dir-files"))
-      case move if move.hasPath("move") =>
-        Move(move.getString("move.source"), move.getString("move.target"))
+      case delete if delete.hasPath(HoconConstants.delete) => Delete(delete.getString(HoconConstants.delete))
+      case mkdir if mkdir.hasPath(HoconConstants.mkDir)    => MakeDir(mkdir.getString(HoconConstants.mkDir))
+      case touchz if touchz.hasPath(HoconConstants.touchz) => Touchz(touchz.getString(HoconConstants.touchz))
+      case chmod if chmod.hasPath(HoconConstants.chmod) =>
+        Chmod(chmod.getString(s"${HoconConstants.chmod}.${HoconConstants.path}"),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The string literal '${ .. }.${ .. }' appears 5 times in the file.](https://app.codacy.com/app/SimonJPegg/scoozie/pullRequest?prid=3072461)